content_scripts/ui_component.coffee

```diff
@@ -8,7 +8,10 @@ class UIComponent
 
   toggleIframeElementClasses: (removeClass, addClass) ->
     @iframeElement.classList.remove removeClass
-    @iframeElement.classList.add addClass
+    unless @iframeElement.classList.contains addClass
+      @iframeElement.classList.add addClass
+      if addClass == "vimiumUIComponentVisible"
+        getComputedStyle(@iframeElement).display
```

Comment:
Pretty sure this is a no-op, and the value doesn't seem to be used at any of the callsites. Is this needed?

Reply:
No, it's not a no-op. `getComputedStyle` will force Chrome to execute a "layout" operation during the calling, and then its inner code can find HUD becomes visible and then send a "child-frame-become-visible" message to the HUD process (i.e. Vimium process). Only when the message is received, Chrome's renderer will accept `element.focus` / `selection.toString()` commands from JavaScript.

A much more detailed tracing has been mentioned in https://github.com/philc/vimium/pull/3277#issuecomment-487363284

Reply:
Okay, makes sense. Might be worth adding a comment to that effect, since it's not obvious from the code alone.

Reply:
Thanks for tips. Added.